<launch>
    <node pkg="sentry_localization" type="pointcloud_to_laserscan_node" name="pointcloud_to_laserscan_node" output="screen">
        <remap from="cloud_in" to="/livox/points"/>
        <rosparam>
            target_frame: livox_mid360 # Leave disabled to output scan in pointcloud frame
            transform_tolerance: 0.01
            min_height: 0.0
            max_height: 0.5
 
            angle_min: -3.1415926 # -M_PI
            angle_max: 3.1415926 # M_PI
            angle_increment: 0.003 # 0.17degree
            scan_time: 0.05
            range_min: 0.35
            range_max: 100
            use_inf: true
            inf_epsilon: 1.0
 
            # Concurrency level, affects number of pointclouds queued for processing and number of threads used
            # 0 : Detect number of cores
            # 1 : Single threaded
            # 2->inf : Parallelism level
            concurrency_level: 1
        </rosparam>
    </node>

    <node pkg="sentry_localization" type="scan_to_scan_filter" name="scan_to_scan_filter_node" output="screen">
        <remap from="scan" to="/scan" />
        <rosparam file="$(find sentry_localization)/config/scan_filter_params/box_filter.yaml" command="load" />
    </node>

</launch>